appengine/backend.py

```diff
@@ -0,0 +1,44 @@
+from abc import abstractmethod, ABCMeta
+
+
+class Backend(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def catalog(self):
+        """Returns an instance of the catalog to use for dataset checking"""
+
+    @abstractmethod
+    def algebra(self):
+        """Returns corresponding target algebra"""
+
+    @abstractmethod
+    def compile_query(self, query, logical_plan, physical_plan, language=None):
+        """Takes the raw query, logical,b and physical plan
+           Returns JSON of compiled query"""
```

Comment:
Just a clarification: Return type is some representation of the query that the backend can process, but not necessarily JSON, right?